src/HealthChecks.AzureStorage/AzureQueueStorageHealthCheck.cs

```diff
@@ -1,79 +1,46 @@
-using System.Collections.Concurrent;
 using Azure.Core;
 using Azure.Storage.Queues;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
 
 namespace HealthChecks.AzureStorage
 {
     public class AzureQueueStorageHealthCheck : IHealthCheck
     {
-        private readonly string? _connectionString;
-        private readonly string? _queueName;
-
-        private readonly TokenCredential? _azureCredential;
-        private readonly Uri? _queueServiceUri;
-
-        private static readonly ConcurrentDictionary<string, QueueServiceClient> _queueClientsHolder = new();
+        private readonly QueueServiceClient _queueServiceClient;
+        private readonly IOptionsSnapshot<QueueStorageHealthCheckOptions> _options;

```

Comment:
`IOptionsSnapshot` usages and overall design look overcomplicated. No one HC from this repo uses `IOptionsSnapshot` and even `IOptions`. I have a strong feeling that it is done so intentionally.

Reply:
Look at other HCs and extension methods that configure them. The preferred way is to use `Action` or `Func` delegates with `IServiceProvider`.

Reply:
I've removed IOptions and used the aforementioned pattern instead